jaxrs-api/src/main/java/jakarta/ws/rs/core/MediaType.java

```diff
@@ -224,19 +224,20 @@ public MediaType() {
         this(MEDIA_TYPE_WILDCARD, MEDIA_TYPE_WILDCARD, null, null);
     }
 
-    private MediaType(final String type, final String subtype, final String charset, Map<String, String> parameterMap) {
+    private MediaType(final String type, final String subtype, final String charset, final Map<String, String> parameterMap) {
 
         this.type = type == null ? MEDIA_TYPE_WILDCARD : type;
         this.subtype = subtype == null ? MEDIA_TYPE_WILDCARD : subtype;
 
-        if (parameterMap == null) {
-            parameterMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+        Map<String, String> map = parameterMap;
```

Comment:
I do not understand why you do this here. It makes the code harder to understand.

Reply:
This is a good suggestion detected by Checkstyle. There was an inconsistent use of final in this method and assignments to parameters are generally frowned upon.